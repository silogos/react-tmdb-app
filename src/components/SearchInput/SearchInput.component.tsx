import { useCallback, useState } from "react";
import { SearchInputProps } from "./SearchInput.type";

function SearchInput({ onSubmit, className }: SearchInputProps) {
  const [searchText, setSearchText] = useState("");
  const handleSubmit = useCallback(
    (evt: React.FormEvent<HTMLFormElement>) => {
      evt.preventDefault();
      onSubmit(searchText);
    },
    [onSubmit, searchText]
  );

  return (
    <form className={className} onSubmit={handleSubmit}>
      <input
        data-testid="search-input"
        name="query"
        type="search"
        className="w-full border-2 rounded-lg  px-4 py-2 bg-white"
        placeholder="Search..."
        onChange={(evt) => setSearchText(evt.target.value)}
        required
      />
    </form>
  );
}

export default SearchInput;
