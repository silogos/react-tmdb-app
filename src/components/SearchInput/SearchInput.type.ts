export type SearchInputProps = {
  className?: string;
  onSubmit: (searchText: string) => void;
};
