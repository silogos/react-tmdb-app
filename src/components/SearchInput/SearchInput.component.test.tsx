import { describe, it, expect, vi } from "vitest";
import SearchInput from "./SearchInput.component";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("SearchInput Component", () => {
  it("renders correctly", () => {
    const component = render(<SearchInput onSubmit={() => {}} />);
    expect(component).toBeTruthy();
  });

  it("displays the correct image source", async () => {
    const user = userEvent.setup();
    const handleSubmit = vi.fn();
    const { getByTestId } = render(<SearchInput onSubmit={handleSubmit} />);
    const inputSearch = getByTestId(
      "search-input"
    ) as unknown as HTMLInputElement;

    expect(inputSearch).toBeInTheDocument();

    inputSearch.focus();
    expect(inputSearch).toHaveFocus();

    await user.keyboard("hello{Enter}");

    expect(handleSubmit).toHaveBeenCalledExactlyOnceWith("hello");
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
