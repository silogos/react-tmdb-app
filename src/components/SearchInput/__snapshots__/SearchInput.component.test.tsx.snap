// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput Component > renders correctly 1`] = `
<div>
  <form>
    <input
      class="w-full border-2 rounded-lg  px-4 py-2 bg-white"
      data-testid="search-input"
      name="query"
      placeholder="Search..."
      required=""
      type="search"
    />
  </form>
</div>
`;
